internal/common/splunk/common.go

```diff
@@ -22,26 +22,36 @@ const (
 	SFxEventCategoryKey   = "com.splunk.signalfx.event_category"
 	SFxEventPropertiesKey = "com.splunk.signalfx.event_properties"
 	SourcetypeLabel       = "com.splunk.sourcetype"
+	SplunkHECTokenHeader  = "Splunk"
+	SplunkHecTokenLabel   = "com.splunk.hec.access_token"
+	// HecEventMetricType is the type of HEC event. Set to metric, as per https://docs.splunk.com/Documentation/Splunk/8.0.3/Metrics/GetMetricsInOther.
+	HecEventMetricType = "metric"
 )
 
+// AccessTokenPassthroughConfig configures passing through access tokens.
 type AccessTokenPassthroughConfig struct {
-	// Whether to associate datapoints with an organization access token received in request.
+	// AccessTokenPassthrough indicates whether to associate datapoints with an organization access token received in request.
 	AccessTokenPassthrough bool `mapstructure:"access_token_passthrough"`
 }
 
-// Metric represents a metric in Splunk HEC format
-type Metric struct {
+// Event represents a metric in Splunk HEC format
+type Event struct {
 	Time       float64                `json:"time"`                 // epoch time
 	Host       string                 `json:"host"`                 // hostname
 	Source     string                 `json:"source,omitempty"`     // optional description of the source of the event; typically the app's name
 	SourceType string                 `json:"sourcetype,omitempty"` // optional name of a Splunk parsing configuration; this is usually inferred by Splunk
 	Index      string                 `json:"index,omitempty"`      // optional name of the Splunk index to store the event in; not required if the token has a default index set in Splunk
-	Event      string                 `json:"event"`                // type of event: this is a metric.
-	Fields     map[string]interface{} `json:"fields"`               // metric data
+	Event      interface{}            `json:"event"`                // type of event: set to "metric" if the event represents a metric.
+	Fields     map[string]interface{} `json:"fields,omitempty"`     // metric data
 }
 
-// GetValues extracts metric key value pairs from a Splunk HEC metric.
-func (m Metric) GetValues() map[string]interface{} {
+// IsMetric returns true if the splunk event is a metric.
+func (m Event) IsMetric() bool {
+	return m.Event == HecEventMetricType
```

Comment:
Does this work even though `m.Event` is an `interface{}`? I'm not confident enough in the subtleties of Go equality to know.

Reply:
I'll test for it.

Reply:
added test with various objects.